### PR TITLE
Update templates to match helm best practices

### PR DESCRIPTION
**Pull request by TomNeyland**
Some tweaks for your pr [#228](https://github.com/CrunchyData/postgres-operator/pull/228) based on the tips in the following links, and on the default service scaffolding helm provides.
https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices
https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/templates.md\#names-of-defined-templates
https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md\#using-rbac-resources

### CODE REVIEW

**Comment by blaggacao** [comment]
Thanks very greateful for this hands-on teaching! :smile:

File: chart/postgres-operator/templates/deployment.yaml [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
     heritage: {{ .Release.Service }}
 spec:
   replicas: {{ .Values.replicaCount }}
   template:
     metadata:
       labels:
-        app: {{ template "name" . }}
+        app: {{ template "postgres-operator.name" . }}
         release: {{ .Release.Name }}
         name: postgres-operator
```
Comment (blaggacao) [inline chart/postgres-operator/templates/deployment.yaml]: Did this slip you view? Or is this supposed to be hard-coded?

File: chart/postgres-operator/templates/service.yaml [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
-  type: "ClusterIP"
+  type: {{ .Values.service.type }}
   ports:
-    - name: apiserver
-      port: 8443
-      protocol: TCP
+    - port: {{ .Values.service.port }}
       targetPort: 8443
+      protocol: TCP
+      name: apiserver
```
Comment (blaggacao) [inline chart/postgres-operator/templates/service.yaml:17]: Is it ok to leave in this simple notation?

File: chart/postgres-operator/templates/service.yaml [hunk 3]
```diff
@@ -1,20 +1,22 @@
...
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
 spec:
-  type: "ClusterIP"
+  type: {{ .Values.service.type }}
   ports:
-    - name: apiserver
-      port: 8443
-      protocol: TCP
+    - port: {{ .Values.service.port }}
```
Comment (blaggacao) [inline chart/postgres-operator/templates/service.yaml:14]: We should probably provide defaults in the `values.yaml`...

**Review by blaggacao — CHANGES_REQUESTED** [review]
The other comments are rather questions, but would you mind to add defaults in the `values.yaml`?

File: chart/postgres-operator/templates/service.yaml (hunk 3 above)
Comment (TomNeyland) [reply chart/postgres-operator/templates/service.yaml:14]: @blaggacao good catch, updated the commit to include the new values.

File: chart/postgres-operator/templates/service.yaml (hunk 2 above)
Comment (TomNeyland) [reply chart/postgres-operator/templates/service.yaml:17]: I am not quite sure what you are asking, could you rephrase the question?

**Comment by TomNeyland** [comment]
Also, these tweaks don't necessarily represent an exhaustive implementation of the suggestions in [chart best practices](https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices), just a few things that caught my eye.

File: chart/postgres-operator/templates/service.yaml (hunk 2 above)
Comment (blaggacao) [reply chart/postgres-operator/templates/service.yaml:17]: Sry for not being clear here:
Shouldn't `name: apiserver` rather be `name: {{ template "postgres-operator.fullname" . }}-apiserver`?

File: chart/postgres-operator/templates/deployment.yaml (hunk 1 above)
Comment (blaggacao) [reply chart/postgres-operator/templates/deployment.yaml]: Shouldn't `name: postgres-operator` rather be `name: {{ template "postgres-operator.fullname" . }}` ?

**Review by blaggacao — APPROVED** [review]
